### PR TITLE
[dynamo] Turn on inline_inbuilt_nn_modules

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131275
* #132053

Known issues that are deliberately kept open and will be fixed later are tracked here - https://github.com/pytorch/pytorch/issues/131696

Training dashboard ([link](https://hud.pytorch.org/benchmark/compilers?dashboard=torchinductor&startTime=Thu%2C%2018%20Jul%202024%2000%3A03%3A50%20GMT&stopTime=Thu%2C%2025%20Jul%202024%2000%3A03%3A50%20GMT&granularity=hour&suite=torchbench&mode=training&dtype=amp&lBranch=gh/anijain2305/435/head&lCommit=408b9358b8fca3a5d08b39741419fe8a596941aa&rBranch=gh/anijain2305/435/base&rCommit=d31f2ae904ba2cf0884bf24413ba2109c3585d51))

image

Inference dashboard ([link](https://hud.pytorch.org/benchmark/compilers?dashboard=torchinductor&startTime=Thu%2C%2018%20Jul%202024%2000%3A03%3A50%20GMT&stopTime=Thu%2C%2025%20Jul%202024%2000%3A03%3A50%20GMT&granularity=hour&suite=torchbench&mode=inference&dtype=bfloat16&lBranch=gh/anijain2305/435/head&lCommit=914244fa2fe0055917e039e35183b21fa90afdc6&rBranch=gh/anijain2305/435/base&rCommit=d31f2ae904ba2cf0884bf24413ba2109c3585d51))
image

Inference sees a little bit more perf degradation but we are ok with that.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

**Review by jansel — APPROVED** [review]

File: torch/_dynamo/config.py [hunk 1]
```diff
@@ -365,9 +365,7 @@ def _get_optimize_ddp_mode():
 enable_cpp_guard_manager = os.environ.get("TORCHDYNAMO_CPP_GUARD_MANAGER", "1") == "1"
 
 # Inline inbuilt nn modules
-inline_inbuilt_nn_modules = (
-    os.environ.get("TORCHDYNAMO_INLINE_INBUILT_NN_MODULES", "0") == "1"
-)
+inline_inbuilt_nn_modules = False if is_fbcode() else True
```
Comment (jansel) [inline torch/_dynamo/config.py:368]:
```suggestion
inline_inbuilt_nn_modules = not is_fbcode()
```

File: torch/_dynamo/config.py [hunk 2]
```diff
@@ -98,7 +98,7 @@ def is_fbcode():
 allow_ignore_mark_dynamic = False
 
 # Set this to False to assume nn.Modules() contents are immutable (similar assumption as freezing)
-guard_nn_modules = False if is_fbcode() else True
+guard_nn_modules = not is_fbcode()
```
Comment (anijain2305) [inline torch/_dynamo/config.py:101]: Just doing a null test to see rocm tests tail. Will revert back the last commit.

**Comment by anijain2305** [comment]
Rocm tests failed on a null-test, where I made a change in config.py that is essentially a noop - https://hud.pytorch.org/pytorch/pytorch/pull/131275?sha=f9e8b3fcda97c726e592484d782c3dc7fe4f14c2

**Pull request closed without merging**